app/src/main/kotlin/com/simplemobiletools/smsmessenger/helpers/MessagesImporter.kt

```diff
@@ -0,0 +1,78 @@
+package com.simplemobiletools.smsmessenger.helpers
+
+import android.content.Context
+import android.net.Uri
+import android.provider.Telephony.*
+import android.util.Log
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import com.simplemobiletools.commons.extensions.queryCursor
+import com.simplemobiletools.commons.extensions.showErrorToast
+import com.simplemobiletools.commons.helpers.ensureBackgroundThread
+import com.simplemobiletools.smsmessenger.extensions.*
+import com.simplemobiletools.smsmessenger.helpers.MessagesImporter.ImportResult.*
+import com.simplemobiletools.smsmessenger.models.ExportedMessage
+import java.io.File
+
+class MessagesImporter(private val context: Context) {
+    enum class ImportResult {
+        IMPORT_FAIL, IMPORT_OK, IMPORT_PARTIAL
+    }
+
+    private val gson = Gson()
+    private val messageWriter = MessagesWriter(context)
+    private val config = context.config
+    private var messagesImported = 0
+    private var messagesFailed = 0
+
+    fun importMessages(path: String, onProgress: (total: Int, current: Int) -> Unit = { _, _ -> }, callback: (result: ImportResult) -> Unit) {
+        ensureBackgroundThread {
+            try {
+
+                val inputStream = if (path.contains("/")) {
+                    File(path).inputStream()
+                } else {
+                    context.assets.open(path)
+                }
+
+                inputStream.bufferedReader().use { reader ->
+                    val json = reader.readText()
+                    val type = object : TypeToken<List<ExportedMessage>>() {}.type
+                    val messages = gson.fromJson<List<ExportedMessage>>(json, type)
+                    val totalMessages = messages.flatMap { it.sms }.size + messages.flatMap { it.mms }.size
```

Comment:
if there are no MMS messages in the file, this gives an error "java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter elements". Or you can also try deselecting "Export MMS" and "Import MMS" too. If you cannot reproduce it, I can send you a sample file.

Reply:
I will look into it.